multiplier_analysis/nbconverted/1_get_multiplier_LV_coverage.py

```diff
@@ -0,0 +1,249 @@
+    
+    generic_cov_i, specific_cov_i = process.get_LV_coverage(
+        generic_genes_processed,
+        specific_genes_processed,
+        multiplier_model_z_processed
+    )
+    
+    generic_cov_ls += list(generic_cov_i)
+    specific_cov_ls += list(specific_cov_i)
+    
+from matplotlib_venn import venn2
+venn2([set(generic_cov_ls), set(specific_cov_ls)],
+      set_labels=("Generic LVs", "Specific LVs")
+     )
+
+
+# In[13]:
+
+
+# Create table of unique generic LVs, unique specific LVs, shared LVs
+process.create_LV_df(generic_cov_ls, specific_cov_ls, multiplier_model_summary)
+
+
+# In[14]:
+
+
+# Find the number of generic and specific genes that have a nonzero contribution to LV
+generic_cov = []
+specific_cov = []
+num_significant_LVs = multiplier_model_z_processed.shape[1]
+
+for ifile in range(len(ls_data_files)):
+    generic_genes = ls_genes_out[ifile][0]
+    specific_genes = ls_genes_out[ifile][1]
+    
+    # Only include those genes that are in multiplier otherwise will get NAs
+    generic_genes_processed, specific_genes_processed = process.process_generic_specific_gene_lists(
+        generic_genes, 
+        specific_genes, 
+        multiplier_model_z_processed
+    )
+    print(len(generic_genes_processed), len(specific_genes_processed))
+    
+    generic_cov_i, specific_cov_i = process.get_LV_coverage(
+        generic_genes_processed,
+        specific_genes_processed,
+        multiplier_model_z_processed
+    )
+    
+    generic_cov.append(len(generic_cov_i)/num_significant_LVs)
+    specific_cov.append(len(specific_cov_i)/num_significant_LVs)
+    
+gene_cov = pd.DataFrame({'Proportion of significantly associated LVs covered': generic_cov + specific_cov,
+                         'gene type': ['generic']*len(ls_data_files) + ['specific']*len(ls_data_files) 
+                      })
+
+
+# In[42]:
+
+
+# Plot coverage distribution given list of generic coverage, specific coverage
+print(generic_cov)
+print(specific_cov)
+
+import textwrap
+fig = sns.boxplot(data=gene_cov, 
+                  x='gene type', 
+                  y='Proportion of significantly associated LVs covered', 
+                  palette=['grey','powderblue'])
+fig.set_xlabel("Gene Type",fontsize=14)
+fig.set_ylabel(textwrap.fill("Proportion of significantly associated LVs covered", width=30),fontsize=14)
+fig.tick_params(labelsize=14)
+fig.set_title("")
+
+
+# In[17]:
+
+
+# Save plot
+fig.figure.savefig(
+        output_figure_filename,
+        format="svg",
+        bbox_inches="tight",
+        transparent=True,
+        pad_inches=0,
+        dpi=300,
+    )
+
+
+# **Takeaway:**
+# * On average, specific genes cover fewer pathway-associated LVs compared to generic genes, which were found to be linked to all pathway-associated LVs.
+# * This difference in coverage is correlated with the fact that there 1-6 specific genes identified compared to the 6000 generic genes found.
```

Comment:
This is a neat analysis! I'm a bit skeptical of drawing conclusions from only 1-6 specific genes, but the results for generic genes (of which there are many more) being associated with many pathways make intuitive sense.

Your box plot looked at how many LVs had at least one generic gene in them. Could you flip this around: for each gene, plot how many LVs it's associated with; i.e. how many LVs that gene is nonzero in? I'm curious what this distribution would look like.

Reply:
Yes I agree with your skeptism completely. Its not surprising that there was more coverage for generic genes vs specific genes because there were many more generic genes found. What if we try to randomly sample a generic gene set that is of the same size as the number of specific genes?

Reply:
After matching the number of generic and specific genes, there is equal coverage of LVs. This also isn't surprising given that most LVs have at least 1000's genes that contribute to them, so the number of genes is making a difference. For the next analysis we will try to account for the amount of contribution of genes in addition to presence/absence